images/controller/internal/controllers/rv-publish-controller/reconciler_test.go

```diff
@@ -0,0 +1,490 @@
+/*
+Copyright 2025 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rvpublishcontroller_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	v1alpha1 "github.com/deckhouse/sds-replicated-volume/api/v1alpha1"
+	"github.com/deckhouse/sds-replicated-volume/api/v1alpha3"
+	rvpublishcontroller "github.com/deckhouse/sds-replicated-volume/images/controller/internal/controllers/rv-publish-controller"
+)
+
+func TestRvPublishReconciler(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "rv-publish-controller Reconciler Suite")
+}
+
+var expectedError = errors.New("test error")
+
+var _ = Describe("Reconcile", func() {
+	scheme := runtime.NewScheme()
+	Expect(v1alpha1.AddToScheme(scheme)).To(Succeed())
+	Expect(v1alpha3.AddToScheme(scheme)).To(Succeed())
+
+	var (
+		builder *fake.ClientBuilder
+		cl      client.WithWatch
+		rec     *rvpublishcontroller.Reconciler
+	)
+
+	BeforeEach(func() {
+		builder = fake.NewClientBuilder().
+			WithScheme(scheme).
+			WithStatusSubresource(&v1alpha3.ReplicatedVolume{})
+		cl = nil
+		rec = nil
+	})
+
+	JustBeforeEach(func() {
+		cl = builder.Build()
+		rec = rvpublishcontroller.NewReconciler(cl, logr.New(log.NullLogSink{}), scheme)
+	})
+
+	It("returns nil when ReplicatedVolume not found", func(ctx SpecContext) {
+		result, err := rec.Reconcile(ctx, reconcile.Request{NamespacedName: client.ObjectKey{Name: "non-existent"}})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result).To(Equal(reconcile.Result{}))
```

Comment:
Одной строчкой делаем

```suggestion
		result, err := rec.Reconcile(ctx, reconcile.Request{NamespacedName: client.ObjectKey{Name: "non-existent"}})
		Expect(err).NotTo(HaveOccurred())
		Expect(result).To(Equal(reconcile.Result{}))
```